### ISSUE TITLE
Date fields aren't being created as dates inside Elasticsearch

### ISSUE THREAD

**Issue by KyleAMathews** [taterbase/mongoosastic#18]
They're being created instead like this:

```
changed: {
    dynamic: true
    type: object
}
```

I'm setting my two date fields like this `created: { type: Date, index: true, es_type:'date' }`

Also, for dates, we should probably also support setting custom date formats per the ElasticSearch documentation - http://www.elasticsearch.org/guide/reference/mapping/core-types.html

**Comment by KyleAMathews** [comment]
This is the mapping that is created within Mongoosastic:

```
{ title: { type: 'string', boost: 2 },
  body: { type: 'string' },
  nid: { type: 'integer' },
  created: { type: 'date' },
  changed: { type: 'date' },
  deleted: { type: 'object' },
  _user: { type: 'string' } }
```

**Comment by KyleAMathews** [comment]
Played around with this for awhile today. The problem, as it turns out is that dates weren't in the ISO 8601 format by default coming from Mongoose. In the serialize method I added a line similar to the following for each date field and dates were then stored correctly in ElasticSearch - `serializedForm.created = new Date(serializedForm.created).toJSON()`

**Comment by jamescarr** [comment]
Fixed as of 2ff2827
